lib/src/workspace.rs

```diff
@@ -175,6 +175,7 @@ impl Workspace {
         op_heads_store_factory: impl FnOnce(&Path) -> Box<dyn OpHeadsStore>,
         index_store_factory: impl FnOnce(&Path) -> Box<dyn IndexStore>,
         submodule_store_factory: impl FnOnce(&Path) -> Box<dyn SubmoduleStore>,
+        workspace_id: WorkspaceId,
```

Comment:
Feel free to annotate this function with `#[allow(clippy::too_many_arguments)]` as far as I'm concerned.

Reply:
That was indeed the root cause. Added.